packages/headless/src/features/attached-results/attached-results-analytics-actions.ts

```diff
@@ -8,20 +9,75 @@ import {
-  makeInsightAnalyticsAction('insight/caseAttach', (client, state) => {
-    validateResultPayload(result);
-    const metadata = getCaseContextAnalyticsMetadata(state.insightCaseContext);
-    return client.logCaseAttach(
-      partialDocumentInformation(result, state),
-      documentIdentifier(result),
-      metadata
-    );
-  })();
+  makeInsightAnalyticsAction({
+    prefix: 'insight/caseAttach',
+    __legacy__getBuilder: (client, state) => {
+      validateResultPayload(result);
+      const metadata = getCaseContextAnalyticsMetadata(
+        state.insightCaseContext
+      );
+      return client.logCaseAttach(
+        partialDocumentInformation(result, state),
+        documentIdentifier(result),
+        metadata
+      );
+    },
+    analyticsType: 'InsightPanel.ItemAction',
+    analyticsPayloadBuilder: (state): InsightPanel.ItemAction => {
+      const metadata = getCaseContextAnalyticsMetadata(
+        state.insightCaseContext
+      );
+      const identifier = documentIdentifier(result);
+      const information = partialDocumentInformation(result, state);
+      return {
+        itemMetadata: {
+          uniqueFieldName: identifier.contentIDKey,
+          uniqueFieldValue: identifier.contentIDValue,
+          title: information.documentTitle,
+          author: information.documentAuthor,
+          url: information.documentUri,
+        },
+        position: information.documentPosition,
+        searchUid: state.search?.response.searchUid || '',
+        action: 'attach',
+        context: {
+          targetId: metadata.caseId || '',
+          targetType: 'Case',
+          caseNumber: metadata.caseNumber,
+        } as InsightPanel.Context,
+      };
+    },
+  });
 
-export const logCaseDetach = (resultUriHash: string) =>
-  makeInsightAnalyticsAction('insight/caseDetach', (client, state) =>
-    client.logCaseDetach(
-      resultUriHash,
-      getCaseContextAnalyticsMetadata(state.insightCaseContext)
-    )
-  )();
+export const logCaseDetach = (result: Result) =>
+  makeInsightAnalyticsAction({
+    prefix: 'insight/caseDetach',
+    __legacy__getBuilder: (client, state) => {
+      return client.logCaseDetach(
+        result.raw.urihash,
+        getCaseContextAnalyticsMetadata(state.insightCaseContext)
+      );
+    },
+    analyticsType: 'InsightPanel.DetachItem',
+    analyticsPayloadBuilder: (state): InsightPanel.DetachItem => {
+      const metadata = getCaseContextAnalyticsMetadata(
+        state.insightCaseContext
+      );
+      const identifier = documentIdentifier(result);
+      const information = partialDocumentInformation(result, state);
+      return {
+        itemMetadata: {
+          uniqueFieldName: identifier.contentIDKey,
+          uniqueFieldValue: identifier.contentIDValue,
+          title: information.documentTitle,
+          author: information.documentAuthor,
+          url: information.documentUri,
+        },
+        context: {
```

Comment:
So we don't have the Case Subject?

Reply:
Exactly, InsightPanel.Context is supposed to have the `CaseSubject`.

However, currently we don't have this information in the state. 

That's what we have in the state:
```
{
    caseContext: {...}
    caseId: '...',
    caseNumber: '...',
  };
``` 

So only the `caseId` and the `caseNumber` are the the information that we sure have.

The case subject could be added in the `caseContext` object, but we have no way of knowing under which key it is.

Additionally we can't add all the `CaseContext` in the metadata of the event, cause any key  sent un the metadata and which is not defined in the schema will strictly make the event invalid.

Unless there is a very powerful reason behind why we need the subject here, I think we should just update the schema by removing the `CaseSubject`

Reply:
@felixpelletier ?